### PR TITLE
Chem Group Recordings

### PR DESCRIPTION
**Pull request by Azrun**
## About the PR 
Reworks the textual copy-paste aspects of chemistry groups to be based on recordings of operations performed at the dispenser.  

## Why's this needed? 
https://forum.ss13.co/showthread.php?tid=15961&pid=176596#pid176596

This shifts chem groups creation and usage to be an actual time investment instead of allowing for a quick copy paste and go.   
The act of manually dispensing the chemicals to generate recordings will also likely improve the general usage of the machine.

## Changelog

```changelog
(u)Azrun
(*)Chem groups have been reworked to be recordings of actions performed.
```

### CODE REVIEW

**Comment by nefarious6th** [comment]
nice. 

this also just makes more sense to me too, imo. when i was first starting out, this is what i THOUGHT the button did.

**Comment by Conchuckter** [comment]
It would be cool if these could be imported/exported as computer files, so you could share them around.

**Comment by TDHooligan** [comment]
it might be worth considering that this technically makes chemgroups more powerful than before (no need to make two chemgroups to handle different steps of a reaction, now you can just make 1)

could also be a bit misleading that it only tracks additions (though tracking removals is probably worse)

still, much better than the current system.

**Comment by Garash2k** [comment]
> This shifts chem groups creation and usage to be an actual time investment instead of allowing for a quick copy paste and go.

That's a very noble goal, but man I'd really like it if it were a fun time investment instead of the plenitude of clicks and numbering input this would make it. `{number input} 4 {carbon} {number input} 22 {hydrogen} {number input} 15 {oxygen}...` 6 clicks and 5 keystrokes and I'm not even halfway through my Pentetic acid chem group, and that's much further away from the complex stimulant or poison groups I've made over so many scientist rounds.

What if the scientist had to run around a bit to get his chem groups? Either going to a dwaine terminal, writing his group somewhere in the mainframe, and having the chem dispensers connected to the mainfraime, being able to download the groups from there. There could be a funny looking "download bar" delay for balance purposes! Or the ThinkDOS way, where the spaceman would have to find a floppy drive, put it in a computer, write his groups on it, then put his disk in the chem dispenser. longer groups could take more size on the floppy!

I have a bit of a horse in this race because my chem note file has more than 100 chem groups, all hand crafted, so I'd be sad if the more complex ones turned into "recipes to follow that take you 30 clicks and 20 keystrokes" But I think my point having the time investment be fun is still worth raising.

File: tgui/packages/tgui/interfaces/ChemDispenser.js [hunk 1]
```diff
@@ -381,22 +398,22 @@ export const ChemGroups = (props, context) => {
...
             onInput={(e, value) => setGroupName(value)} />
-          <Box pt={1} as="span">
+          {/* <Box pt={1} as="span">
             <Input
               pl={5}
               placeholder="Reagents"
               value={reagents}
               onInput={(e, value) => setReagents(value)} />
-          </Box>
+          </Box> */}
```
Comment (ZeWaka) [inline tgui/packages/tgui/interfaces/ChemDispenser.js:416]: delete the commented out sections

**Review by ZeWaka — CHANGES_REQUESTED** [review]

**Pull request merged**